snow/consensus/snowman/poll/early_term_no_traversal.go

```diff
@@ -222,6 +275,75 @@ func (p *earlyTermNoTraversalPoll) PrefixedString(prefix string) string {
 	)
 }
 
-func (p *earlyTermNoTraversalPoll) String() string {
+func (p *earlyTermTraversalPoll) String() string {
 	return p.PrefixedString("")
 }
+
+func aggregateVotesFromPrefixesAndIDs(transitiveVotesForPrefixes map[string]int, transitiveVotes bag.Bag[ids.ID]) []int {
+	voteCountsForIDsOrPrefixes := make([]int, 0, len(transitiveVotesForPrefixes)+len(transitiveVotes.List()))
+
+	for _, id := range transitiveVotes.List() {
+		votesForID := transitiveVotes.Count(id)
+		voteCountsForIDsOrPrefixes = append(voteCountsForIDsOrPrefixes, votesForID)
+	}
+
+	for _, voteCount := range transitiveVotesForPrefixes {
+		voteCountsForIDsOrPrefixes = append(voteCountsForIDsOrPrefixes, voteCount)
+	}
+	return voteCountsForIDsOrPrefixes
+}
+
+func computeTransitiveVotesForPrefixes(votesGraph *voteGraph, transitiveVotes bag.Bag[ids.ID]) map[string]int {
+	votesForPrefix := make(map[string]int)
+	votesGraph.traverse(func(v *voteVertex) {
+		descendantIDs := descendantIDsOfVertex(v)
+		pg := longestSharedPrefixes(descendantIDs)
+		// Each shared prefix is associated with a bunch of IDs.
+		// Sum up all the transitive votes for these blocks,
+		// and return all such shared prefixes indexed by the underlying transitive descendant IDs.
+		pg.bifurcationsWithCommonPrefix(func(ids []ids.ID) {
+			key := concatIDs(ids)
+			count := sumVotesFromIDs(ids, transitiveVotes)
+			votesForPrefix[key] = count
+		})
+	})
+	return votesForPrefix
```

Comment:
Here we create `votesForPrefix` as a map of `[]ids.ID -> count`, and then we later drop the `[]ids.ID` and just use `count`. We never read from `votesForPrefix`. Why are we creating this map? It seems like we could at least just use a slice for this... Alternatively, we could just use the counts as we calculate them.

Reply:
Yeah makes sense, addressed [here](https://github.com/ava-labs/avalanchego/pull/3337/commits/93d2779439569eab9ee4df946f06c6579bdee4dc).